### PR TITLE
Add "Remarks/Maoni" Section in HPS client Services #338

### PR DESCRIPTION
**Pull request by fhonsiah**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] or Closes #[issue number]

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have built and run the Client's app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the Client app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: opensrp-chw-hps/src/main/java/org/smartregister/chw/hps/interactor/BaseHpsServiceVisitInteractor.java [hunk 1]
```diff
@@ -90,7 +108,21 @@ private void evaluateHpsClientCriteria(Map<String, List<VisitDetail>> details) t
...
+                        new AppExecutors().mainThread().execute(() -> callBack.preloadActions(actionList));
+                        evaluateHpsRemarks(details);
+                    } catch(Exception e){
+                        Timber.e(e.toString());
+                    }
+                    return BaseHpsVisitAction.Status.COMPLETED;
+                }
+                return BaseHpsVisitAction.Status.PENDING;
+            }
+        };
```
Comment (cozej4) [inline opensrp-chw-hps/src/main/java/org/smartregister/chw/hps/interactor/BaseHpsServiceVisitInteractor.java:125]: This part is not required. HpsClientCriteriaActionHelper is enough to achieve the intended action.
Ukiiweka hii part na pia kwenye other action helpers, contents za Comment Action zitakuwa overriden kila wakati mtu akijaza any other action

File: opensrp-chw-hps/src/main/java/org/smartregister/chw/hps/interactor/BaseHpsServiceVisitInteractor.java [hunk 2]
```diff
@@ -102,7 +134,21 @@ private void evaluateHpsEducationOnBehavioralChange(Map<String, List<VisitDetail
...
```
Comment (cozej4) [inline opensrp-chw-hps/src/main/java/org/smartregister/chw/hps/interactor/BaseHpsServiceVisitInteractor.java:151]: This part is not required.

File: opensrp-chw-hps/src/main/java/org/smartregister/chw/hps/interactor/BaseHpsServiceVisitInteractor.java [hunk 3]
```diff
@@ -114,7 +160,21 @@ private void evaluateOtherHpsServices(Map<String, List<VisitDetail>> details) th
...
```
Comment (cozej4) [inline opensrp-chw-hps/src/main/java/org/smartregister/chw/hps/interactor/BaseHpsServiceVisitInteractor.java:177]: This part is not required.

File: opensrp-chw-hps/src/main/java/org/smartregister/chw/hps/interactor/BaseHpsServiceVisitInteractor.java [hunk 4]
```diff
@@ -126,7 +186,21 @@ private void evaluateCurativeServices(Map<String, List<VisitDetail>> details) th
...
```
Comment (cozej4) [inline opensrp-chw-hps/src/main/java/org/smartregister/chw/hps/interactor/BaseHpsServiceVisitInteractor.java:203]: This part is not required.

File: opensrp-chw-hps/src/main/java/org/smartregister/chw/hps/interactor/BaseHpsServiceVisitInteractor.java [hunk 5]
```diff
@@ -136,6 +210,19 @@ private void evaluateReferralServices(Map<String, List<VisitDetail>> details) th
         actionList.put(context.getString(R.string.hps_referral_services), action);
     }
 
+    private void evaluateHpsRemarks(Map<String, List<VisitDetail>> details) throws BaseHpsVisitAction.ValidationException {
+
+        HpsRemarksActionHelper actionHelper = new HpsRemarksActionHelper(mContext, memberObject);
+        BaseHpsVisitAction action = getBuilder(context.getString(R.string.hps_remarks))
+                .withOptional(false)
```
Comment (cozej4) [inline opensrp-chw-hps/src/main/java/org/smartregister/chw/hps/interactor/BaseHpsServiceVisitInteractor.java:217]: this should be `.withOptional(true)` since the comments section is optional and the user should be able to save even without filling any comments

**Review by cozej4 — CHANGES_REQUESTED** [review]

**Review by cozej4 — APPROVED** [review]
